ui/src/controls/ToggleDarkModeButton.svelte

```diff
@@ -0,0 +1,32 @@
+<svelte:head>
+{#if stored == "dark"}
+<link rel='stylesheet' href='darkmode.css' id="dark-mode-css">
+{/if}
+</svelte:head>
+
+<script>
+import { createEventDispatcher } from 'svelte';
+const dispatch = createEventDispatcher();
+var stored = localStorage.getItem("dark-mode-storage");
+
+if (!stored) {
+  if (window.matchMedia("(prefers-color-scheme: dark)").matches ) {
```

Comment:
Would we want to listen to the changes? I'm not sure if it's a valid use case, but could imagine iOS to automatically switch themes based on time or sensor input, and I guess a user could expect the site to change without reloading: https://developer.mozilla.org/en-US/docs/Web/API/MediaQueryList/addListener

You'd have to make sure to de-register the listener, though, once the theme is switched manually. And there is no way back to "system theme" once we switched once... What do you think?

Reply:
The only way to have both the option to switch automatically and remember a manual selection would be with an additional button that allows going back. I think an additional button would counteract the minimalist design principles of the app.